src/coreclr/jit/target.h

```diff
@@ -208,8 +208,156 @@ enum _regMask_enum : unsigned
 // In any case, we believe that is OK to freely cast between these types; no information will
 // be lost.
 
-#if defined(TARGET_AMD64) || defined(TARGET_ARMARCH) || defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
+typedef _regNumber_enum regNumber;
+typedef unsigned char   regNumberSmall;
+
+#if defined(TARGET_AMD64) || defined(TARGET_ARM) || defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
 typedef unsigned __int64 regMaskTP;
+#elif defined(TARGET_ARM64)
+typedef unsigned __int64 regMaskSmall;
+struct regMaskTP
+{
+    unsigned __int64 low;
+
+    regMaskTP(unsigned __int64 regMask)
+        : low(regMask)
+    {
+    }
+
+    regMaskTP()
+    {
+    }
+
+    FORCEINLINE explicit operator bool() const
+    {
+        return low != 0;
+    }
```

Comment:
We should go through and replace these with `!= RBM_NONE` at some point, instead of having this.